R/deploySite.R

```diff
@@ -102,3 +91,65 @@ deploySite <- function(siteDir = getwd(),
     ...
   )
 }
+
+quartoSite <- function(path, quiet = FALSE, error_call = caller_env()) {
+  check_installed(
+    "quarto",
+    reason = "to deploy quarto sites",
+    call = error_call
+  )
+
+  config <- quarto::quarto_inspect(path)$config
+
+  list(
+    render = function() quarto::quarto_render(path, quiet = quiet, as_job = FALSE),
+    name = basename(normalizePath(path)),
+    title = config$website$title %||% config$book$title %||% config$title,
+    output_dir = outputDir(path, config$project$`output-dir`)
+  )
+}
+
+rmarkdownSite <- function(siteDir, quiet = FALSE, error_call = caller_env()) {
+  check_installed(
+    "rmarkdown",
+    version = "0.9.5.3",
+    reason = "to deploy RMarkdown sites",
+    call = error_call
+  )
+
+  # discover the site generator
+  siteGenerator <- rmarkdown::site_generator(siteDir)
+  if (is.null(siteGenerator)) {
+    cli::cli_abort(
+      "No `_site.yml` found in {.path {siteDir}.}",
+      call = error_call
+    )
+  }
+
+  list(
+    render = function() {
```

Comment:
How different is this from calling `rmarkdown::render_site`?

The `render_site` function returns the name of the output file, which is generally `output/index.html` or `./index.html`, and the output directory can be derived from it, leaving a render function that returns its output location looking something like:

```r
render = function() {
    result <- rmarkdown::render_site(...)
    return(dirname(result))
}
```

Reply:
I don't know, I just copied the existing code 😄. If you think it's likely to work, I'm happy to switch (and I'll re-run the appropriate tests).

Reply:
It's quite similar to `render_site`, but I don't think it's worth the bother. Mostly, I was leaning into the idea that the `render()` could return the `output-dir` rather than computing it separately (since it is only necessary when rendering).